### ISSUE TITLE
Always treat parachain that never produced block for a significant amount of time as unlocked.

### ISSUE THREAD

**Issue by xlc** [paritytech/polkadot#7539]
Right now one a parathread is promoted to parachain, it is locked and only the parachain sovereign origin can unlock it.

This can be an issue if the parachain wasm/head is broken and can never produce block. Governance is required to fix this.

Alternatively, we can just automatically unlock it if it is unable to produce block for say 7 days. Then the para manager can fix the wasm/head.

**Comment by bkchr** [comment]
This should be RFC before implementing this change.

**Comment by bkchr** [comment]
Sounds like a reasonable idea. However, it also makes the lock avoidable.

We also need to think on how this makes sense with the new scheduling system. (This alone puts up a lot of questions that need thinking)

**Comment by bkchr** [comment]
As said in element. This is more a phylosophical question and not a technical one. From technical side it makes sense and solves the problem at hand. The technical solution also would be really easy.

**Comment by xlc** [comment]
Yeah. This is only a short term solution. We will need to revisit this when parathead is ready.
